### PR TITLE
generate listers with upstream lister-gen

### PR DESCRIPTION
**Pull request by MHBauer**
I'm checking this in because it's what I can get.

in theory tagging the unversioned types with the `// +genclient=true` annotation should also generate the `internalversion` typed listers, but it does not for me. This should also require the addition of the package `github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog` to the `--input-dirs` flag. Even with all of that in place, unversioned listers are not generated. I will follow up with the creators of the generators.

### CODE REVIEW

**Comment by arschles** [comment]
This is great for the versioned generators. LGTM

**Comment by MHBauer** [comment]
`groupName=` annotation seems to need to exist, but it doesn't have any effect on the output.

**Comment by MHBauer** [comment]
we were missing several pieces to get the internal version to generate. I'm not sure what we're missing to get the internal client. More research is needed.

File: Makefile [hunk 1]
```diff
@@ -133,6 +137,13 @@ $(BINDIR)/client-gen:
 		--clientset-path github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/ \
 		--go-header-file vendor/github.com/kubernetes/repo-infra/verify/boilerplate/boilerplate.go.txt
 	  # the previous three directories will be changed from kubernetes to apimachinery in the future
+# generate lister
```
Comment (duglin) [inline Makefile]: indent, and below too

**Comment by duglin** [comment]
Just 2 minor "indent" fixes

File: Makefile (hunk 1 above)
Comment (MHBauer) [inline Makefile]: Why? My makefile linter flags them as issues.

Comment (duglin) [inline Makefile]: Things should align. Perhaps you're using spaces or tabs in place of the other?

Comment (MHBauer) [inline Makefile]: I always use tabs in makefiles, but apparently that's not standard either. If you really want it, I'll put up with bright neon pink warnings.

**Comment by duglin** [comment]
LGTM

ping @pmorie

**Comment by pmorie** [comment]
Let's get the `groupName` fields correct.  LGTM otherwise

File: Makefile [hunk 2]
```diff
@@ -138,6 +142,14 @@ $(BINDIR)/client-gen:
 		--clientset-path github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/ \
 		--clientset-name clientset \
 		--go-header-file vendor/github.com/kubernetes/repo-infra/verify/boilerplate/boilerplate.go.txt
+	  # the previous three directories will be changed from kubernetes to apimachinery in the future
```
Comment (pmorie) [inline Makefile]: lose this line, must have come in with the rebase

Comment (MHBauer) [inline Makefile]: ai-ya

**Pull request merged**